services/121-service/src/shared/services/custom-http.service.ts

```diff
@@ -254,7 +254,7 @@ export class CustomHttpService {
    * @param data - Any key-value object
    * @returns - A copy of the input-object with some specific data overwritten/redacted
    */
-  private redactSensitiveDataProperties(data: any) {
+  private redactSensitiveDataProperties<T extends object>(data: T): T {
```

Comment:
Hmm. But `data` is sometimes also “only a string”…

Would that create a runtime-issue?


Or would we need to add a `| number | string | boolean` somewhere?

Reply:
> Hmm. But data is sometimes also “only a string”…

I would've expected a compiler error in that case, but now I see that the payloads coming in are also typed as `any` so that doesn't help lol

> Would that create a runtime-issue?

No, it wouldn't. I created a [TS playground](https://www.typescriptlang.org/play/?#code/GYVwdgxgLglg9mABAJwKYBMCG0DKqwDOMsAbqgCKZSYAKycADqsrKgQDwAqiqAHlPnQFEcAEYArVNAB8ACizUAXIk4BKZdwDeAKESIICAlBQZsA9JWqIAvIk2IAdE4WZEAXwDciAPTfEOAAtMABtguAB3fUYAT0QoOERMEjgYdEQAWxBqWDAAcziA1BFkGFyYMBCRCSkobV1EGGBEWXq9AHIQAmYK9NQ2hqQ0LGgMS1cAMnHWuOimOCahs1GqTAdO7sxem2tbNqMSvLb61TtpxZGLFbWu5B6i3YI4XqgA8ty2j3q3Or00KBBbogXJ9vtooLMigBZaIAGWIUGCqAASqgAI4gNjGWw6PQMTDRMKYdDKTBgaIguraAyEYzIZTQuFQBHItEYow2U64-GE4mIPZPVAvN5HUHUx6IhxhXKyNqcTFtVSfMVwCVS2TnKB4QjEGBkMZ0RjMVgEdUOPEEuBE1SKoA) to mimic this behaviour. But, I still don't like the way this is typed now.

> Or would we need to add a | number | string | boolean somewhere?

Potentially, yes. But I'd rather remove `extends object` from the generic instead. I've pushed a change that takes this into account.